### PR TITLE
Better components and documentation

### PR DESCRIPTION
**Pull request by lucadealfaro**
This implements better components, with an url() methods so that they can be instantiated from vue.  I have also added quite a bit of documentation to the components.  Also, the vue grid has a search bar, and the table fields are sortable.

### CODE REVIEW

File: apps/examples/components/README.md [hunk 1]
```diff
@@ -32,28 +30,133 @@ For example, one can instantiate a component via:
...
+
 Once the component is thus defined, it can be used in a controller, for example via:
 
     @action('mypage')
     @action.uses(session)
     def mypage():
         c = my_component(id=1)
         return dict(my_component=my_component)
-        
+
```
Comment (agavgavi) [inline apps/examples/components/README.md:42]: Where is c being used here? The line below mentions `my_component(id=1)` which is set to c. Should the return instead be `return dict(my_component=c)` ?

Comment (lucadealfaro) [reply apps/examples/components/README.md]: Thanks, typo fixed.

**Comment by lucadealfaro** [comment]
I have added sortable fields in tables as well.

**Comment by mdipierro** [comment]
Really really nice. Not sure we should promise backward compatibility on this quite yet as we may want more feedback from users.

**Comment by lucadealfaro** [comment]
Yes, you are right, I think we should add a note saying that this is not fully stable yet. 
But I find these components are shaping up to be nicely useful.

**Pull request merged**